### PR TITLE
Checks for dataContext before applying FallbackValue to explicit bindings

### PR DESCRIPTION
**Pull request by NicolasChampagne**
## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
iOS TimePicker resets binding to default value

## What is the new behavior?
Bindings with datacontext dont reset to defaultvalue

## PR Checklist

Please check if your PR fulfills the following requirements:
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

Internal Issue (If applicable):
https://nventive.visualstudio.com/Umbrella/_workitems/edit/156571

### CODE REVIEW

File: src/Uno.UI.Tests/BinderTests/Given_Binder.cs [hunk 1]
```diff
@@ -865,6 +865,22 @@ public void When_SelfRelativeSource()
 			Assert.AreEqual(42, SUT.Property2);
 		}
 
+#if !NET461
```
Comment (davidjohnoliver) [inline src/Uno.UI.Tests/BinderTests/Given_Binder.cs]: The unit tests are `NET461` only... Why do you have this condition?

Comment (NicolasChampagne) [reply src/Uno.UI.Tests/BinderTests/Given_Binder.cs]: The explicit method binding that gets triggered is !Net461 only

Comment (NicolasChampagne) [reply src/Uno.UI.Tests/BinderTests/Given_Binder.cs]: guess I can't unittest it.

File: src/Uno.UI/DataBinding/BindingExpression.cs [hunk 2]
```diff
@@ -115,14 +115,17 @@ Binding binding
...
+				ApplyFallbackValue();
+			}
+			
 			ApplyExplicitSource();
 			ApplyElementName();
 		}
 
 		private ManagedWeakReference GetWeakDataContext()
-			=> _isElementNameSource || _explicitSourceStore.IsAlive ? _explicitSourceStore : _dataContext;
+			=> _isElementNameSource || (_explicitSourceStore?.IsAlive ?? true) ? _explicitSourceStore : _dataContext;
```
Comment (davidjohnoliver) [inline src/Uno.UI/DataBinding/BindingExpression.cs]:
```suggestion
			=> _isElementNameSource || (_explicitSourceStore?.IsAlive ?? false) ? _explicitSourceStore : _dataContext;
```

File: src/Uno.UI.Tests/BinderTests/Given_Binder.cs (hunk 1 above)
Comment (davidjohnoliver) [reply src/Uno.UI.Tests/BinderTests/Given_Binder.cs]: What exactly is missing for this? It can be added to `NET461`. The databinding layer should be fully coverable by unit tests.

File: src/Uno.UI.Tests/BinderTests/Given_Binder.cs [hunk 3]
```diff
@@ -884,7 +884,7 @@ public BaseTarget(BaseTarget parent = null)
 				DataContextChangedCount++;
 			}
 
-			#region ChildrenBinders DependencyProperty
+#region ChildrenBinders DependencyProperty
```
Comment (jeromelaban) [inline src/Uno.UI.Tests/BinderTests/Given_Binder.cs]: This seems like unrelated changes ? Did you mean to add a unit test here ?

Comment (NicolasChampagne) [reply src/Uno.UI.Tests/BinderTests/Given_Binder.cs]: Yes I did however the test i wanted to add tests a method not available in net461. I left the autoformatted code changes

**Comment by NicolasChampagne** [comment]
> This change also requires a unit test. The UI Test demonstrates a use case, but the behavior can be replicated in a platform agnostic way.

@jeromelaban the unitTest has been added

File: src/Uno.UI.Tests/BinderTests/Given_Binder.cs [hunk 4]
```diff
@@ -848,6 +848,21 @@ public void When_ExplicitUpdateSourceTrigger_UpdateSource()
...
 		}
 
+		[TestMethod]
+		public void When_ExplicitSetBindingBetweenProperties_IsNotFallBackValue()
+		{
+			var source = new MyBindingSource { IntValue = 42 };
+			var target = new MyControl();
+			var target2 = new MyObjectTest();
+
+			target.SetBinding(MyControl.MyPropertyProperty, new Binding { Source = source, Path = nameof(MyBindingSource.IntValue), Mode = BindingMode.TwoWay, UpdateSourceTrigger = UpdateSourceTrigger.Explicit });
```
Comment (jeromelaban) [inline src/Uno.UI.Tests/BinderTests/Given_Binder.cs]: You change is making explicit changes with regards to weak references; you should ensure that the source is actually garbage collected to validate the scenario.
